packages/react-components/react-teaching-popover-preview/src/components/TeachingPopoverCarousel/Carousel/CarouselItem/useCarouselItem.ts

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
+import { CarouselItemProps, CarouselItemState } from './CarouselItem.types';
+import { useCarouselContext_unstable } from '../CarouselContext';
+
+export const useCarouselItem_unstable = (
+  props: CarouselItemProps,
+  ref: React.Ref<HTMLDivElement>,
+): CarouselItemState => {
+  const { value } = props;
+  const visible = useCarouselContext_unstable(c => c.value === value);
+
+  return {
+    value,
+    visible,
+    components: {
+      root: 'div',
+    },
+    root: slot.always(
+      getIntrinsicElementProps('div', {
+        ref,
+        'data-carousel-item': value,
+        'data-carousel-active-item': visible,
+        hidden: !visible,
+        ...props,
+      }),
+      { elementType: 'div' },
+    ),
+  };
```

Comment:
We need to ensure that we don't render `children` when an item is not visible (currently it renders contents of all items):

```ts
if (!visible) {
  state.root.children = null;
}
```